conans/client/rest/rest_client_v2.py

```diff
@@ -39,20 +39,24 @@ def _get_file_list_json(self, url):
     def get_recipe(self, ref, dest_folder, metadata, only_metadata):
         url = self.router.recipe_snapshot(ref)
         data = self._get_file_list_json(url)
-        files = data["files"]
-        accepted_files = ["conanfile.py", "conan_export.tgz", "conanmanifest.txt", "metadata/sign"]
-        if only_metadata:
-            accepted_files = []
-        metadata = metadata or []
-        metadata = [f"metadata/{m}" for m in metadata]
-        files = [f for f in files if any(f.startswith(m) for m in accepted_files)
-                 or any(fnmatch.fnmatch(f, m) for m in metadata)]
-
-        # If we didn't indicated reference, server got the latest, use absolute now, it's safer
-        urls = {fn: self.router.recipe_file(ref, fn) for fn in files}
-        self._download_and_save_files(urls, dest_folder, files, parallel=True)
-        ret = {fn: os.path.join(dest_folder, fn) for fn in files}
-        return ret
+        server_files = data["files"]
+        result = {}
+
+        if not only_metadata:
+            accepted_files = ["conanfile.py", "conan_export.tgz", "conanmanifest.txt",
+                              "metadata/sign"]
+            files = [f for f in server_files if any(f.startswith(m) for m in accepted_files)]
+            # If we didn't indicated reference, server got the latest, use absolute now, it's safer
+            urls = {fn: self.router.recipe_file(ref, fn) for fn in files}
+            self._download_and_save_files(urls, dest_folder, files, parallel=True)
+            result.update({fn: os.path.join(dest_folder, fn) for fn in files})
+        if metadata:
```

Comment:
We need to separate the download to define ``metadata=True`` that allows 1) not caching and 2) allows overwriting existing files